Modification affichage offres

2265857 opened the pull request:
# Description

Liste à cocher pour t'aider à faire une revue de code efficace.

## Qualité du code

- [x] Les noms des variables, fonctions et classes sont descriptifs et suivent les conventions de nommage du projet. Ex: `calculateTotalPrice()` vs `calc()` ou `userFirstName` vs `ufn`.
- [x] Le code qui se répète doit être groupé à un endroit pour être réutilisé facilement.

## Tests

- [x] Les tests testent les cas valides
- [x] Les tests testent tous les cas fautifs

## Documentation

- [x] La documentation a été mise à jour si nécessaire

## Gestion des erreurs

- [x] On gère toutes les erreurs. Ex: (pas de `catch` vide ou de `return;` en cas d'erreur)
- [x] On utilise le logger pour journaliser les erreurs

## Quand tu as terminé

lancer coderabbitai en ajoutant `@Coderabbitai` dans ta revue de code.

Linkky committed:
Visuel coloré sur les nom des catégorie

Linkky committed:
Merge branch 'dev' of https://github.com/ecoleduweb/EmploiEtudiant into Administrateur-Ajout-NomEntreprise-&-Date-création

Linkky committed:
Modification de la vue pour l'administrateur:
• Amélioration et restructuration du html des <div> sont maintenant dans une <table> pour un meilleur visuel.
• Alignement des colonnes avec les entête.
• Modification du css

Linkky committed:
Nettoyage de code inutile

Linkky committed:
Merge branch 'dev' of https://github.com/ecoleduweb/EmploiEtudiant into Modification-Affichage-Offres

Linkky committed:
Modification de l'affichage de Titre par Poste visé de la page emplois

Linkky committed:
ajutement de la vue mobile

Linkky committed:
Nettoyage du code & modification du X du modale
• Ajout handleOverlayClick qui permet de fermer le modal lorsque l'utilisateur click à l'extérieur du modal.

• Ajout de disableOverlayClose pour éviter de fermer le modal lorsque non souhaité exemple lors de la création d'offre d'emploi dans le dashboard quand le user clique sur Button .Créer une nouvelle offre.

Linkky committed:
enlever disableOverlayClose

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -6,30 +6,93 @@
     }
+
+    // Ferme le modal si l'utilisateur clique à l'extérieur du modal
+    const handleOverlayClick = (event: MouseEvent) => {
```
À retirer, ce n'est pas demandé par le client.

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -6,30 +6,93 @@
+    }
+
+    // Ferme le modal si l'utilisateur appuie sur la touche Escape
+    const handleKeyDown = (event: KeyboardEvent) => {
```
À retirer, ce n'est pas demandé par le client.

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -6,30 +6,93 @@
-        <button class="close" on:click={handleButtonClick}>
-            <img src="cancel.svg" class="image" alt="close" />
-        </button>
+<div class="modal-container">
```
Il ne faut pas mettre le x en bas. Il reste à la même place,en haut, mais il ne fait pas qu'il ait une colonne pout lui tout seul, on perd quelques pixes pour afficher de l'information. J'ai mis une capture d'écran pour t'aider à comprenre, n'hésite pas si tu as encore une question

![image](https://github.com/user-attachments/assets/a9f3f88e-2623-42bb-a6de-45a57b256de9)

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -6,30 +6,93 @@
+    }
+
+    /* Animation d’apparition de l'overlay */
+    @keyframes fadeIn {
```
Pas d'animation svp.

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -6,30 +6,93 @@
+    }
+
+    /* Animation d’ouverture du modal */
+    @keyframes scaleIn {
```
Pas d'animation.

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -58,4 +121,10 @@
         height: 30px;
     }
+
+    .image:hover {
```
pas demandé

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
 
-    .info-mobile {
-        display: none;
+    tr:hover {
```
Pas nécessaire (à retirer)

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
+        
+        /* Style pour les cellules mobiles */
+        .mobile-content, .mobile-employer, .mobile-details {
+            display: table-cell; /* Assure que les cellules sont traitées comme des cellules de tableau */
```
c'est utile?

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
+    <!-- Section mobile-->
+    <td class="mobile-content">{offer.title}</td>
+    <td class="mobile-employer">{offer.enterprise?.name}</td>
+    <td class="mobile-details"><img class="image" src="add.svg" alt="ajouter" /></td>
```
SImplement utiliser la balise image plus bas comme elle est toujours affichée.

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
-</button>
+<tr class="offreEmploi" on:click={() => handleModalClick(offer)}>
+    <!-- Section mobile-->
+    <td class="mobile-content">{offer.title}</td>
```
Comme on affiche toujours le titre et le nom de l'entreprise, je crois qu'on peut laisser tomber la classe mobile.

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
+    <td class="info">{offer.deadlineApply}</td>
+    <td class="info">{offer.studyPrograms?.map(x => x.name).join(', ')}</td>
+    <td class="info">{offer.enterprise?.name}</td>
+    <td><img class="info image" src="add.svg" alt="ajouter" /></td>
```
retirer la class info (pour n'avoir besoin que d'une balise td pour le +)

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
     @media (max-width: 768px) {
-        .info {
+        /* Si vous souhaitez que seules les 3 premières cellules soient visibles */
+        .offreEmploi td:nth-child(n+4) {
```
Je crois que c'est plus simple de garder la class info avec le display: none puisque si jamais on veut afficher un troisième champ ou qu'on veut mettre le nom de l'entreprise à la fin, on va être coincé à cause de la css. d'ailleurs, on va en profiter pour changer la classe info pour hide-on-mobile? Ce serait plus clair. (parce que info...... c'est pas clair haha)

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
+    @media (min-width: 769px) {
+        /* En mode desktop, cachez les cellules mobiles */
+        .mobile-content, .mobile-employer, .mobile-details {
+            display: none;
```
À retirer quand on va avoir changer la logique d'affichage

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -34,44 +42,20 @@
+        align-items: center;
+        color: white;
+        /*width: 55%; */
+        width: 100%; /* ajouté pour tester nouveau model d'affichange */
```
À Valider après le changement de code.

antoineCegepRDL replied:
Retirer le 55% en commentaire si on garde le 100%

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -83,28 +67,15 @@
     }
-    .button:hover {
-        background-color: #455b66;
+
```
Ne pas modifier le desing actuel, juste changer l'allignement.

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -175,58 +172,112 @@
         <section class="offres">
             <h1 class="title">
-                <span class="text">MES OFFRES D'EMPLOIS </span>
+                <span>MES</span>
```
Remettre comme avant.

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -175,58 +172,112 @@
-                        handleArchiveModalClick={() => {handleArchiveClick(offer)}}
-                    />
-                {/each}
+                <h2 class="textSections">Offres <span class="hightlight-red">refusées</span></h2>
```
Remettre comme avant svp

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -175,58 +172,112 @@
             {#if toBeApprovedOffer.length > 0}
-                <h2 class="textSections">Offres en attente d'approbation</h2>
-                {#each toBeApprovedOffer as offer}
+            <h2 class="textSections">Offres en <span class="hightlight-orange">attente d'approbation</span></h2>
```
Remettre comme avant svp

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -175,58 +172,112 @@
             {/if}
             {#if expiredOffer.length > 0}
-                <h2 class="textSections">Offres expirées</h2>
+                <h2 class="textSections">Offres <span class="hightlight-red">expirées</span></h2>
```
Remettre comme avant svp

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -341,9 +394,38 @@
+        table-layout: fixed; /* Force une répartition égale des colonnes */
+    }
+
+    thead {
```
Retirer le gradient et toute la css qui ne vient pas directement avoir d'impact sur l'allignement svp

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -341,9 +394,38 @@
+    }
+
     @media (max-width: 768px) {
+        h2 {
```
remettre l'ancienne valeur

antoineCegepRDL commented on front-end/projet_application/src/routes/emplois/+page.svelte:
```diff
@@ -69,21 +69,24 @@
+            <table>
+                <thead>
+                    <tr>
+                        <th class="mobile-emploi">Poste visé</th>
```
good catch ici

antoineCegepRDL commented on front-end/projet_application/src/routes/emplois/+page.svelte:
```diff
@@ -158,10 +148,26 @@
+    }
+
+    thead {
+        background: linear-gradient(2deg, hsl(218, 27%, 16%, 50%),hsl(173, 100%, 34%, 20%), hsl(173, 100%, 34%, 30%), hsl(173, 100%, 34%, 20%), hsl(218, 27%, 16%, 50%));
```
À retirer svp

antoineCegepRDL commented on front-end/projet_application/src/routes/dashboard/+page.svelte:
```diff
@@ -175,58 +172,112 @@
+                <h2 class="textSections">Offres <span class="hightlight-red">refusées</span></h2>
+                <table>
+                    <thead>
+                        <tr>
```
Comme on a toujours les mêmes titres et les mêmes rangés, on va se créer un component en svelte qui va contenir : les titres et les rangés. Comme ça, si jamais on change un titre ou qu'on affiche un truc de plus, on va faire le changement à un seul endroit! :)

antoineCegepRDL replied:
Bref, ce que je veux dire c'est de créer un component avec la table et on va la réutiliser selon les sections.

antoineCegepRDL replied:
Utiliser le componsant de la table ici

JeanFelixLarouche commented on front-end/projet_application/src/routes/emplois/+page.svelte:
```diff
@@ -158,10 +162,27 @@
-        display: none;
+
+    thead {
+        background: linear-gradient(2deg, hsl(218, 27%, 16%, 50%),hsl(173, 100%, 34%, 20%), hsl(173, 100%, 34%, 30%), hsl(173, 100%, 34%, 20%), hsl(218, 27%, 16%, 50%));
```
Simplement une question : Pourquoi utiliser HSLA plutôt que RGBA ou les hexa ? Y a-t-il un avantage ?

2265857 replied:
HSLA permet une meilleur contrôle sur la teinte, la saturation et la luminosité. Les dégradés en HSLA ont tendance à être plus naturels, car ils respectent une progression logique de la teinte et de la luminosité.


Comparativement à RGBA qui à chaque fois que l'on désire éclaircir ou assombrir une couleur dans un dégradé, il faudra recalculer les trois valeurs R, G et B à chaque fois.

HSLA:
background: linear-gradient(2deg, hsl(218, 27%, 16%, 50%),hsl(173, 100%, 34%, 20%), hsl(173, 100%, 34%, 30%), hsl(173, 100%, 34%, 20%), hsl(218, 27%, 16%, 50%));
RGBA:
background: linear-gradient(2deg, rgba(30, 38, 52, 0.5),rgba(0, 173, 153, 0.2), rgba(0, 173, 153, 0.3), rgba(0, 173, 153, 0.2), rgba(30, 38, 52, 0.5));

JeanFelixLarouche commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,70 @@
     @media (max-width: 768px) {
-        .info {
+        /* Si vous souhaitez que seules les 3 premières cellules soient visibles */
+        .offreEmploi td:nth-child(n+4) {
```
Peut-être que c'est la bonne façon de faire mais si tu ajoutes ou enlèves une cellule dans le tableau il faudra aussi modifier le css pour correspondre avec le nombre. Est-ce qu'on ne pourrait pas faire comme lorsqu'on cache les éléments mobiles en utilisant leur classe pour modifier leur affichage ?

2265857 replied:
Oui, tu as tout à fait raison, c'est important de rendre le code le plus dynamique possible à l'aide d'une classe.
J'ai apporté les corrections nécessaire.

JeanFelixLarouche reviewed (commented):
J'ai mis 2 commentaires/questions pour des trucs dont je n'étais pas certain.

Linkky committed:
Ajout d'une classe .desktop-only dans la DetailOfferRow.svelte afin de faciliter la gestion d'affichage entre mobile et desktop

Linkky committed:
Correctifs selon la demande d'antoine du pullrequest
https://github.com/ecoleduweb/EmploiEtudiant/pull/300/files/dd134c7e310f5519882915e0df8b3f208270fc8a

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,56 @@
+    <!-- Section mobile-->
+    <td class="mobile-content">{offer.title}</td>
+    <td class="mobile-employer">{offer.enterprise?.name}</td>
+    <td class="mobile-details"><img class="image" src="add.svg" alt="ajouter" /></td>
```
Tu peux garder une seule balise td avec l'image du + vu qu'elle est utilisée dans les deux formats et qu'elle est de la même grosseur.

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,56 @@
-    .info-mobile {
-        display: none;
+    tr:hover {
+        background-color: #555b66;
```
Il manque la transition `        transition: background-color 0.3s ease;` qui était appliquée pour les rangés.

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,56 @@
+    <td class="mobile-details"><img class="image" src="add.svg" alt="ajouter" /></td>
+
+    <!-- Section desktop-->
+    <td class="desktop-only">{offer.title}</td>
```
Comme on affiche toujours le titre de l'offre et l'employeur, on peut retirer la section du haut entièrement et retirer l'affichage conditionnel de mobile-**

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,56 @@
+        .mobile-content, .mobile-employer, .mobile-details {
+            display: none;
+        }
+    }
```
Il resterait à remettre les titres en bleu svp : 
![image](https://github.com/user-attachments/assets/6949c5d5-8c46-4c2c-bef8-0e636ba2515a)

antoineCegepRDL replied:
Pour la grosseur du texte, ce n'était pas une demande, mais vu la quantité d'offre, on va garder ça comme ça,

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -83,28 +67,15 @@
     }
-    .button:hover {
+
+    tr:hover {
```
Attention, il y a un bug avec la vue mobile :
![image](https://github.com/user-attachments/assets/63dd83b0-cf15-4177-a517-686451120796)

antoineCegepRDL replied:
à gauche l'original, à droite ton travail. C'est sur la page d'acceuil que ça cause problème

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -20,13 +20,15 @@
+        position: relative;
         display: flex;
-        flex-direction: row;
+        flex-direction: column;
```
vraiment mieux!

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -1,28 +1,36 @@
-    export let handleEditModalClick: (id: number) => void
-    export let handleApproveModalClick: (id: number) => void
-    export let handleArchiveModalClick: (id: number) => void
+    import { onMount } from "svelte";
```
Dans la page d'administration, la largeur est aussi cassée : 
![image](https://github.com/user-attachments/assets/b77681ff-3948-4cb5-9363-6e628e26fba8)

antoineCegepRDL replied:
En vue desktop

antoineCegepRDL commented on front-end/projet_application/src/routes/emplois/+page.svelte:
```diff
@@ -178,18 +183,23 @@
-            margin-top: 0;
-            margin-bottom: 1vh;
+        
+        .mobile-emploi {
```
Je crois que ces balises ne servent à rien je me trompe? Par contre, si tu veux mettre les colonnes d'un certain % pour l'affichage mobile, tu peux le faire sans déclarer de classe directement dans l'affichage, mais en utilisant des pseudo-classe comme ceci : `tr:nth-child(1) {width:70%}`

antoineCegepRDL commented on front-end/projet_application/src/styles/global.css:
```diff
@@ -42,3 +42,15 @@ h6 {
     color: #00ad9a;
 }
+
+.hightlight-red {
```
Retirer les balises inutiles

Linkky committed:
Correctifs selon la demande (Antoine)

Linkky committed:
Intégration du composant pour les futures tables et ajout du composant table dans la page d'accueil

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Table.svelte:
```diff
@@ -0,0 +1,138 @@
+  import DetailOfferRow from "../JobOffer/DetailOfferRow.svelte";
+
+  // Définition des propriétés du composant
+  export let columns = []; // Tableau contenant la définition des colonnes
```
Je sais que c'est tentant, mais faire un tableau générique, ça cause toujours full de problème. J'aimerais que tu crées un tableau juste pour les offres dans le tableau de bord. On va donc y mettre les titres des colonnes, les rangés, etc... bref, vraiment descendre la complexité de ton composant pour qu'il ne soit utilisable que pour notre besoin.

Aussi, plusieurs éléments ici ne fonctionnent plus du tout avec svelte5 alors c'est aussi un des raisons qui me pousse à demander le changement.

Linkky committed:
Correctifs apporté selon les demandes voir github

Linkky committed:
Retirer le composant Table
Retirer code dans la page Accueil, j'ai testé le composant TableEmplois et il fonctionne bien.
Le code sera alors ajouté dans la page d'Accueil via le Ticket #301
Ajout du composant TableEmplois plus simplifié

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/TableEmplois.svelte:
```diff
@@ -0,0 +1,63 @@
+        import DetailOfferRow from "./DetailOfferRow.svelte";
+
+        export let offers: JobOfferDetails[] = [];
+        export let onOfferClick: (id: number) => void;
```
le nom serait handleOfferClick

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/DetailOfferRow.svelte:
```diff
@@ -28,114 +28,48 @@
     }
 
+    @media (min-width: 769px) {
+    }
```
Balise inutile

antoineCegepRDL commented on front-end/projet_application/src/Components/Common/Modal.svelte:
```diff
@@ -20,13 +20,15 @@
+
         width: 80%;
-        max-width: 800px;
+        max-width: 600px;
```
On pourrait laisser 800 pour la vue desktop non? Ce serait plus agréable

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -1,28 +1,36 @@
+    let enterpriseName = "Entreprise inconnue";
+
+    onMount(() => {
+        if ("enterprise" in offer && offer.enterprise) {
```
if(offer.enterprise) suffit, pas besoin du in

antoineCegepRDL commented on front-end/projet_application/src/Components/JobOffer/OfferRow.svelte:
```diff
@@ -83,28 +66,15 @@
+        background-color: #555b66;
     }
 
     @media (max-width: 768px) {
```
À retrer si ce n'est plus utilisé

antoineCegepRDL commented on front-end/projet_application/src/routes/emplois/+page.svelte:
```diff
@@ -15,11 +16,13 @@
-        selectedOffer = offer
-
-        pushState("?id=" + offer.id, {})
+    const handleAddJobOfferClick = (id: number) => {
```
Passer un jobOffer comme avant, pas un id

Linkky committed:
Merge branch 'Modification-Affichage-Offres' of https://github.com/ecoleduweb/EmploiEtudiant into Modification-Affichage-Offres

Linkky committed:
correctif

Linkky committed:
Ajout du composant TableEmploi sur la page d'accueil est ajustement avec const handleOfferClick pour conserver la logique de redirection vers la page.
handleEmploiWithId utilise un offer.id et handleOfferClick attend un objet offer: JobOfferDetails

Linkky committed:
envelé du code en commentaire

Linkky committed:
modification de la du nom du composant pour TableOffer

Linkky committed:
Merge branch 'dev' of https://github.com/ecoleduweb/EmploiEtudiant into Modification-Affichage-Offres

PlamediIlunga reviewed (approved):
Tout est fonctionnel . Composant bien réutilisé
